pkg/resources/aws/aws.go

```diff
@@ -2102,6 +2103,68 @@ func ListIAMInstanceProfiles(cloud fi.Cloud, clusterName string) ([]*resources.R
 	return resourceTrackers, nil
 }
 
+func ListIAMOIDCProviders(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(awsup.AWSCloud)
+
+	var providers []*string
+	{
+		request := &iam.ListOpenIDConnectProvidersInput{}
+		response, err := c.IAM().ListOpenIDConnectProviders(request)
+		if err != nil {
+			return nil, fmt.Errorf("error listing IAM OIDC Providers: %v", err)
+		}
+		for _, provider := range response.OpenIDConnectProviderList {
+			arn := provider.Arn
+			descReq := &iam.GetOpenIDConnectProviderInput{
+				OpenIDConnectProviderArn: arn,
+			}
+			_, err := c.IAM().GetOpenIDConnectProvider(descReq)
+			if err != nil {
+				return nil, fmt.Errorf("error getting IAM OIDC Provider: %v", err)
+			}
+			// TODO: only delete oidc providers if they're owned by the cluster.
```

Comment:
If the issuer url was 1:1 with the cluster though, I think we could be reasonably sure?

Reply:
the issuer url is 1:1 with the cluster, but we only have access to the cluster name here, and the initial plan was to require the user host the json files themselves at the url, so the url may be totally unrelated to the cluster name even if they are 1:1.

I was imagining a common pattern would be to use an externally-managed public s3 bucket, making the issuer url something like `https://s3.amazonaws.com/peters-oidc-provider-bucket/cluster-foobar/`. It's unique to the cluster but if we only have the cluster name i dont think we can guarantee that we've identified a provider to delete unless we document additional requirements like "the url's path must contain the full cluster name".

I proposed some alternative implementations we could explore down the road in #8264. I'm aware of https://github.com/kubernetes/enhancements/issues/1393 and i think it would be good to integrate, perhaps in this case kops could setup an `oidc` dns record within the cluster's hosted zone that points to a dedicated load balancer targeting the new apiserver endpoints. If we can use that as the issuer url, it not only lets us identify the issuer purely from the cluster name but also lets us compute both the issuer url and the CA thumbprint ourselves so the user doesn't need to provide either. This setup would require an internet-accessible load balancer and the cluster using a public hosted zone, but I think thats a reasonable requirement. Is this what you had in mind?